### ISSUE TITLE
Switching the entire system at once

### ISSUE THREAD

**Issue by Gazgiza** [sharpjd/GPUPrefSwitcher#4]
Hi ! I've been tweaking the program for the past few days but I can't find a way to automatically switch the entire system at once on batery saver mode. I do get some better battery life but not as excepted.
Is there a way to solve this ?
Have a nice day !

**Comment by Gazgiza** [comment]
Hi, I'm using a dell g16 with a 4070 and a 13900hx. And my goal is sort of shutting down the 4070 when unplugged so that every app use the igpu and not having to manually add every app. I've seen sketchy tutorials talking about disabling the driver but it seems more risky than useful. 
Downclocking the cup seems to also be an option but I'll look at it later on.

**Comment by Gazgiza** [comment]
For the moment, the program works for the specified apps but it hasn't turned the gpu off

**Comment by sharpjd** [comment]
Unfortunately there could be a lot of reasons the is remaining on. In my personal experience, these include:
- Not restarting the PC or killing ALL dGPU apps after unplugging the monitor
- Apps that actually use the GPU without your knowledge like MSI Afterburner
- NVIDIA services running in the background and pinging the dGPU frequently (for reasons that are also incomprehensible to me)

Additionally, make sure your laptop's mux switch isn't in dedicated graphics only mode.

The best way to check if there's still anything running is to go to Task Manager -> Details -> Left click column and select "Dedicated GPU Memory" and sort apps by that column. It should be 0 all the way down.

And indeed, disabling the driver or device from Device Manager is *not* the way to go.

**Comment by Gazgiza** [comment]
Are apps like NVIDIA control panel the problem or is it more GEFORCE experience ?

**Comment by BluSalient** [comment]
I've heard that GeForce Experience can cause problems, but that hasn't been the case for me. I personally disable NVIDIA services entirely while on battery using the Task Scheduler trigger function because the dGPU seems to get pinged less. Here's a batch script to automate that (also make a copy but with `net start` instead of `net stop` for when you plug in):

```
echo This script disables NVIDIA related services.
TIMEOUT /T 60
net stop /y NvContainerLocalSystem
net stop /y NVDisplay.ContainerLocalSystem
net stop /y FvSvc
```

**Comment by sharpjd** [comment]
I've heard that GeForce Experience can cause problems, but that hasn't been the case for me. You could just close it and find out. I personally disable NVIDIA services entirely while on battery using the Task Scheduler trigger function because the dGPU seems to get pinged less. Here's a batch script to automate that (also make a copy but with net start instead of net stop for when you plug in):

echo This script disables NVIDIA related services.
TIMEOUT /T 60
net stop /y NvContainerLocalSystem
net stop /y NVDisplay.ContainerLocalSystem
net stop /y FvSvc

**Comment by Gazgiza** [comment]
That's good for me, I haven't installed it.

**Comment by Gazgiza** [comment]
How much W/H is the battery on your laptop and how long does it last ?

**Comment by sharpjd** [comment]
93Wh. When it was in good health and the dGPU was off, 5-7 hours. Power consumption was 10w< idle and up to 30 when browsing the web. 

You can follow tips from my previous comments and find out what your idle power draw is.

**Comment by Gazgiza** [comment]
With HWmonitor ?
> Additionally, make sure your laptop's mux switch isn't in dedicated graphics only mode.
What do you mean by "mux" ?

**Comment by Gazgiza** [comment]
On hw monitor I get around 30watts of consumption on idle and my battery is 86 W/H

**Comment by sharpjd** [comment]
> With HWmonitor ?
> 
> > Additionally, make sure your laptop's mux switch isn't in dedicated graphics only mode.
> 
> What do you mean by "mux" ?

If you do not know what a laptop "Mux Switch" is, I'd highly recommend reading up on it; because if your laptop is directly outputting video from the dGPU to the internal display, then the dGPU will obviously never turn off and keep consuming power.

> On hw monitor I get around 30watts of consumption on idle and my battery is 86 W/H

Turn off background apps and switch away from the High Performance power plan. I have no idea if the 13900HX is just that power hungry; but for reference, my 11800h can consume as low as 1W when idle (the rest is taken by other system components such as the display, keyboard, motherboard, and RAM), adding up to 8-12W total — which is why I think 30W is still a bit unusually high.

**Comment by Gazgiza** [comment]
I think I found the problem. I haven't reduced the max frequency so even though it is at 10%, it still is at 3.5 GHz and more. 
I'll see what I can find about mux. 
Thanks

**Comment by Gazgiza** [comment]
That's wierd : explorer.exe is using multiple mo of dedicated memory and my browser too even though they are supposed to be on battery  saving mode

**Comment by sharpjd** [comment]
Check the Graphics Options panel to make sure that there aren't any apps set to "High Performance"

I also forgot to mention that HWInfo64 also wakes up the GPU when it starts, and may ping it every time you tab in and out. There is also a setting to not wake sleeping GPUs.

It's good you found the apps running on the dGPU. For now, simply try killing all those apps. Remember that unplugging from an external monitor will probably also require to do that.

Chrome sometimes holds some dGPU memory for me while I'm plugged in even without an external monitor. Closing and reopening usually does the trick

**Comment by Gazgiza** [comment]
![Capture d'écran 2024-09-06 082006](https://github.com/user-attachments/assets/6f12392c-0f30-4aea-9cd6-c67b988fd9a1)

**Comment by Gazgiza** [comment]
I have this whereas I should have the option for igpu right ?

**Comment by sharpjd** [comment]
Yes. The "Power Saving" option should show the name of your integrated graphics, not your 4070.

Perhaps your Mux Switch is set to bypass the iGPU. Check your laptop's OEM control panel app to change this. 

If you can't find it there (unlikely since your laptop is new), check the BIOS.

**Comment by Gazgiza** [comment]
> Perhaps your Mux Switch is set to bypass the iGPU. Check your laptop's OEM control panel app to change this.

I found the cause for this : I had disabled nvidia optimus so now I have the option. The dedicated memory usage is now at 0 but the shared memory is still used a bit so my dgpu doesn't go to sleep mode but the power draw is 4 watts less only. Is there a way to disable shared memory ?
